app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivityViewModel.kt

```diff
@@ -29,23 +32,42 @@ class MainActivityViewModel
 @Inject constructor(
     private val playbackManager: PlaybackManager,
     userManager: UserManager,
-    val settings: Settings,
+    private val settings: Settings,
     private val endOfYearManager: EndOfYearManager,
     private val multiSelectBookmarksHelper: MultiSelectBookmarksHelper,
     private val podcastManager: PodcastManager,
     private val bookmarkManager: BookmarkManager,
     private val theme: Theme,
 ) : ViewModel() {
+    private val _state = MutableStateFlow(State())
+    val state = _state.asStateFlow()
 
     var isPlayerOpen: Boolean = false
     var lastPlaybackState: PlaybackState? = null
     val shouldShowStoriesModal = MutableStateFlow(false)
     var waitingForSignInToShowStories = false
 
     init {
+        showWhatsNewIfNeeded()
         updateStoriesModalShowState(!settings.getEndOfYearModalHasBeenShown())
     }
 
+    private fun showWhatsNewIfNeeded() {
+        viewModelScope.launch {
+            val lastSeenVersionCode = settings.getWhatsNewVersionCode()
+            val migratedVersion = settings.getMigratedVersionCode()
+            if (migratedVersion != 0) { // We don't want to show this to new users, there is a race condition between this and the version migration
+                val whatsNewShouldBeShown = WhatsNewFragment.isWhatsNewNewerThan(lastSeenVersionCode)
```

Comment:
I might be misunderstanding but won't this show the WhatsNewFragment with every app upgrade? Rather than only showing the auto play what's new for a certain release?

Reply:
Once `WhatsNewFragment` is shown, we set `WhatsNewVersionCode` in settings here:

```
fun onWhatsNewShown() {
    settings.setWhatsNewVersionCode(Settings.WHATS_NEW_VERSION_CODE)
     _state.update { state -> state.copy(shouldShowWhatsNew = false) }
}
```

So on next upgrade, `WhatsNewFragment.isWhatsNewNewerThan(lastSeenVersionCode)` is `false` and the fragment is not shown.